lib/src/tasks/gen_test_runner/api.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2015 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+library dart_dev.src.tasks.gen_test_runner.api;
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:dart_dev/src/tasks/gen_test_runner/config.dart';
+import 'package:dart_dev/src/tasks/task.dart';
+
+Future<GenTestRunnerTask> genTestRunner(
+    {String directory,
+    Environment environment,
+    String filename,
+    bool genHtml,
+    bool react,
+    List<String> scriptTags}) async {
+  var executable = 'gen-test-runner';
```

Comment:
i kind of like keeping this in the task so it's more easily discerned which command generated which output when running the task